text_recognizer/data/iam_synthetic_paragraphs.py

```diff
@@ -56,11 +58,14 @@ def prepare_data(self, *args, **kwargs) -> None:
     def setup(self, stage: str = None) -> None:
         rank_zero_info(f"IAMSyntheticParagraphs.setup({stage}): Loading trainval IAM paragraph regions and lines...")
 
-        if stage == "fit" or stage is None:
-            line_crops, line_labels = load_line_crops_and_labels("train", PROCESSED_DATA_DIRNAME)
+        def _load_dataset(split: str, transform: ParagraphStem) -> BaseDataset:
+            line_crops, line_labels = load_line_crops_and_labels(split, PROCESSED_DATA_DIRNAME)
             X, para_labels = generate_synthetic_paragraphs(line_crops=line_crops, line_labels=line_labels)
             Y = convert_strings_to_labels(strings=para_labels, mapping=self.inverse_mapping, length=self.output_dims[0])
-            self.data_train = BaseDataset(X, Y, transform=self.trainval_transform)
+            return BaseDataset(X, Y, transform=transform)
```

Comment:
keeping this reformating so we can easily add `self.data_val` if we want. 
In fact, I did add it to a run but a different bug made the run unsuccessful.

Reply:
done at https://github.com/full-stack-deep-learning/fsdl-text-recognizer-2022/pull/31/files#diff-cd205268606d2b02590586c359105225d4cffe942d0e76f9e4c50d33dd260e17R76